funsor/contract.py

```diff
@@ -77,50 +79,22 @@ def eager_subs(self, subs):
                         self.reduced_vars)
 
 
-@optimize.register(Contract, ATOMS[1:], ATOMS, frozenset)
-@optimize.register(Contract, ATOMS, ATOMS[1:], frozenset)
-@eager.register(Contract, ATOMS[1:], ATOMS, frozenset)
-@eager.register(Contract, ATOMS, ATOMS[1:], frozenset)
-def contract_ground_ground(lhs, rhs, reduced_vars):
-    result = _simplify_contract(lhs, rhs, reduced_vars)
-    if result is not None:
-        return result
-
+@optimize.register(Contract, Funsor, Funsor, frozenset)
```

Comment:
This now applies to `Binary`, `Unary`, `Reduce` etc. Should I add a `COMPOUNDS` list with a reflect op as I've done for `Tensor` below?

Reply:
I would say to just remove that line since `optimize` defaults to reflection